docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,201 @@
+    "mariadb": { 
+      "type": "object",
+      "properties": {
+        "enabled": { "type": "boolean", "title": "Enable External Database" }
+      }
+    },
+    "externalDatabase": {
+      "type": "object",
+      "title": "External Database Details",
+      "properties": {
+        "host": { "type": "string" },
+        "user": { "type": "string" },
+        "password": { "type": "string" },
+        "database": { "type": "string" },
+        "port": { "type": "integer" }
+      }
+    },
+    "persistence": {
+      "type": "object",
+      "properties": {
+        "size": {
+          "type": "string",
+          "title": "Disk Size",
+              "anyOf": [
+            {"pattern": ".*"},
+            {"enum": ["10Gi", "50Gi", "100Gi"]},
+          ]
+        }
+      }
+    },
+    "resources": {
+      "type": "object",
+      "title": "Required Resources",
+      "description": "Deployment resource requests",
+      "properties": {
+        "requests": {
+          "type": "object",
+          "properties": {
+            "memory": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["128Mi", "512Mi", "1Gi"] }
+              ]
+            },
+            "cpu": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["100m", "300m", "500m"] }
+              ]
+            }
+          }
+        },
+      }
+    },
+    "enableMetrics": { "type": "boolean", "title": "Enable Metrics" },
+    "replicas": { "type": "integer", "title": "Number of Replicas"}
+  },
+
+  "required": ["wordpressUsername", "wordpressEmail", "persistence", "resources"],
+}
+```
+
+This `values.schema.json` is a valid JSON schema for the `values.yaml` of WordPress. It can be more complete since it's possible to include every single parameter of the values.yaml file.
+
+In order to be able to provide a richer presentation of the form we are including the file `values.form.json`. This allows us to know which parameters we need to represent in the form and how to map them to valid values. In case there is a parameter that is not known we should be able to parse it as a generic input.
+
+```json
+{
+  "username": "wordpressUsername",
+  "password": "wordpressPassword",
+  "email": "wordpressEmail",
+  "enableExternalDatabase": "mariadb.enabled",
+  "externalDatabase": "externalDatabase",
+  "disk": "persistence.size",
+  "resources": "resources",
+  "enableMetrics": "enableMetrics",
+  "replicas": "replicas"
+}
+```
+
+## Deployment Workflow
+
+This section defines how the deployment logic will work based on the JSON Schema definition above.
+
+1. The deployment component loads the details of the Chart. If the chart doesn't contain the file `values.schema.json` or the file `values.form.json`, continue as today; Only the advanced deployment is allowed. In other case, go to the next step.
```

Comment:
Sounds reasonable for the first version. We can check how these files are embraced by the community :+1: